### ISSUE TITLE
Fider Sidebar icons

### ISSUE THREAD

**Issue by mrdaveanderson** [objective-see/WhatsYourSign#6]
Greetings, 

Since 1.2.x or so, every time I install WYS my sidebar icons for the generic HDDs change to circles, but only in the finder sidebar. Anytime it should have an image for a non removable drive, it shows up as a circle in the sidebar now.

I attached 3 images: sidebar before installing WYS, sidebar after installing WYS, and image of drives listed inside my computer (aka the finder version of /volumes/ )

![WYS Beore](https://user-images.githubusercontent.com/19983864/59155098-43b03800-8a36-11e9-9294-9e66bc2e3ae3.png)

![WYS After](https://user-images.githubusercontent.com/19983864/59155097-43b03800-8a36-11e9-9803-e3b323bad9ad.png)

![Volumes](https://user-images.githubusercontent.com/19983864/59155096-43b03800-8a36-11e9-85e0-4c1debc97a5d.png)

Anyone have ideas about anything to look into to see why? I don't have any other finder extensions...nor anything else I can think of that modifies the behavior of the finder that WYS could conflict with.

Thanks,
-Dave

**Comment by strafer** [comment]
I've stopped using WhatsYourSign? for this very reason. Such an annoying bug. And it's not good because the bug has stopped me from using what is a very useful program that enhances the security of my Macs.

Two years without an update and over six months since this issue was filed here. I really hope an update is coming.

**Comment by strafer** [comment]
Version 1.6.0 still seems to corrupt external volume icons in the Finder sidebar. However, uninstalling it doesn't seem to resolve the issue. I don't quite understand what What'sYourSign is modifying in the Finder that would cause it to corrupt the sidebar icons and only for external volumes (both attached external hard drives or network volumes).

**Comment by strafer** [comment]
So, since the dev doesn't bother to respond to issues posted here, does anyone else know how to restore the Finder sidebar icons for external devices that have been corrupted by What's Your Sign? Prior to version 1.6, uninstalling the app fixed the issue. However, with version 1.6, the icons remain corrupted on both my Macs even after uninstalling What's Your Sign and rebooting. 

I've deleted the Finder preferences, the various caches including icon services, but the sidebar icons are still corrupted. The icons look normal in the Finder; it's only the Finder sidebar that they're corrupted.

**Comment by objective-see** [comment]
Sorry for the delayed response. I've been digging to try find more a solution, but with no luck. 

This is almost certainly a bug in macOS. 
First, WhatsYourSign doesn't attempt to directly change or modify icons in any manner. 

However, Apple's [documentation](https://developer.apple.com/library/archive/documentation/General/Conceptual/ExtensibilityPG/Finder.html) says: "The sidebar icons you provide will replace the default folder icons in the Finder’s sidebar for any of the root folders that your extension is monitoring." 

...of course we don't want to replace the icons, so we don't provide any. 
However, as you noted macOS seems to go ahead and modifies them anyways? 

I've honestly spent several days on this, with 0 progress :( 
...the only "solution" appears to not to provide the right-click "Signing Info" for those locations. 

I've found that uninstalling and re-launching Finder.app (option+click -> Relaunch) for me, resets the icons. 

The fact that there seem to be caching issue even following an uninstall and reboot, again points to a core issue in macOS.

**Comment by strafer** [comment]
Thanks for responding.

It's a frustrating problem. That it is most likely a macOS bug does not surprise me given the general decline of Apple's OS and software.

After reading the install/uninstall sequence of WhatsYourSign, I unchecked all the Finder extensions and quick actions I have installed and then relaunched the Finder. That restored the icons. I checked and unchecked them one by one, and it appears that this app, [Service Station](https://servicestation.menu), was somehow preventing the Finder sidebar icons from being restored after I uninstalled What'sYourSign and rebooted.

Now, I don't have Service Station installed on my other Mac and the Finder sidebar icons still display incorrectly on that one, too. But unchecking the What'sYourSign Finder extension and then relaunching the Finder does restore the icons.

As you suggest, it appears to be a bug. However, it's interesting that other Finder extensions from Dropbox or Resilio Sync, for example, don't seem to trigger the bug, so I'm not sure what the triggering factor could be. 

To summarize, WhatsYourSign seems to trigger a Finder sidebar icon display bug. Uninstalling WhatsYourSign or unchecking its Finder extension and then relaunching the Finder restores the correct default Finder icons. In addition, the Finder extension belonging to the app Service Station prevented the corrupted Finder icons from being restored even after WhatsYourSign was uninstalled or its Finder extension disabled.

**Comment by objective-see** [comment]
Thanks for the additional info!

Hrmm, maybe I can look into Dropbox, to see how they were handling this issue :)

**Comment by OothecaPickle** [comment]
any updates on this?

**Comment by strafer** [comment]
Any progress on this issue? The app is still a no-go, unfortunately, until this annoying cosmetic issue is fixed. Unless Apple fixed the Finder extension icon bug in Catalina or Big Sur. I really don't want to upgrade to Catalina because it's shit. But I might be willing to if they fixed the bug.

**Comment by OothecaPickle** [comment]
problem still exists in v2.0.0

**Comment by FrostyGelato** [comment]
Would it be possible to have WYS display its own copy of the default disk icon? Apple is unlikely to ever fix this issue and the blank file icon is really ugly.
